src/lib.rs

```diff
@@ -285,6 +288,25 @@ impl WalkDir {
         self.opts.max_open = n;
         self
     }
+
+    /// Set a compare function.
+    /// If a compare function is set, WalkDir will return all paths in sorted
+    /// order.
+    ///
+    /// ```rust,no-run
+    /// use std::cmp;
+    /// use std::ffi::OsString;
+    /// use walkdir::WalkDir;
+    ///
+    /// fn string_compare(a: &OsString, b: &OsString) -> cmp::Ordering {
+    ///     a.cmp(b)
+    /// }
+    /// WalkDir::new("foo").sort(Some(string_compare));
+    /// ```
+    pub fn sort(mut self, sorter: Option<fn(&OsString,&OsString) -> std::cmp::Ordering>) -> Self {
```

Comment:
Perhaps there is code that allows creating objects by chaining, but avoid calling the same function twice. Such code could be an example. E.g.
```rust
    let a = A::new().color(blue).color(blue);
```
would not compile because `color()` is called twice.
Such code would also pass out objects that implement certain builder traits.